embedded-graphics/src/coord.rs

```diff
@@ -2,184 +2,230 @@
-    pub struct Coord(pub CoordPart, pub CoordPart);
-
-    impl Coord {
-        /// Create a new coordinate with X and Y values
-        pub fn new(x: CoordPart, y: CoordPart) -> Self {
-            Coord(x, y)
-        }
-
-        /// Clamp coordinate components to positive integer range
-        ///
-        /// ```
-        /// # use embedded_graphics::coord::Coord;
-        /// #
-        /// let coord = Coord::new(-5, 10);
-        ///
-        /// assert_eq!(coord.clamp_positive(), Coord::new(0, 10));
-        /// ```
-        pub fn clamp_positive(&self) -> Self {
-            Coord::new(self.0.max(0), self.1.max(0))
-        }
-
-        /// Remove the sign from a coordinate
-        ///
-        ///
-        /// ```
-        /// # use embedded_graphics::coord::Coord;
-        /// #
-        /// let coord = Coord::new(-5, -10);
-        ///
-        /// assert_eq!(coord.abs(), Coord::new(5, 10));
-        /// ```
-        pub fn abs(&self) -> Self {
-            Coord::new(self.0.abs(), self.1.abs())
-        }
+    /// assert_eq!(coord.abs(), Coord::new(5, 10));
+    /// ```
+    pub fn abs(&self) -> Self {
+        Coord::new(self.0.abs(), self.1.abs())
     }
+}
 
-    impl Add for Coord {
-        type Output = Coord;
+impl Add for Coord {
+    type Output = Coord;
 
-        fn add(self, other: Coord) -> Coord {
-            Coord::new(self.0 + other.0, self.1 + other.1)
-        }
+    fn add(self, other: Coord) -> Coord {
+        Coord::new(self.0 + other.0, self.1 + other.1)
     }
+}
 
-    impl AddAssign for Coord {
-        fn add_assign(&mut self, other: Coord) {
-            self.0 += other.0;
-            self.1 += other.1;
-        }
+impl AddAssign for Coord {
+    fn add_assign(&mut self, other: Coord) {
+        self.0 += other.0;
+        self.1 += other.1;
     }
+}
 
-    impl Sub for Coord {
-        type Output = Coord;
+impl Sub for Coord {
+    type Output = Coord;
 
-        fn sub(self, other: Coord) -> Coord {
-            Coord::new(self.0 - other.0, self.1 - other.1)
-        }
+    fn sub(self, other: Coord) -> Coord {
+        Coord::new(self.0 - other.0, self.1 - other.1)
     }
+}
 
-    impl SubAssign for Coord {
-        fn sub_assign(&mut self, other: Coord) {
-            self.0 -= other.0;
-            self.1 -= other.1;
-        }
+impl SubAssign for Coord {
+    fn sub_assign(&mut self, other: Coord) {
+        self.0 -= other.0;
+        self.1 -= other.1;
     }
+}
 
-    impl Index<usize> for Coord {
-        type Output = CoordPart;
+impl Index<usize> for Coord {
+    type Output = i32;
 
-        fn index(&self, idx: usize) -> &CoordPart {
-            match idx {
-                0 => &self.0,
-                1 => &self.1,
-                _ => panic!("Unreachable index {}", idx),
-            }
+    fn index(&self, idx: usize) -> &i32 {
+        match idx {
+            0 => &self.0,
+            1 => &self.1,
+            _ => panic!("Unreachable index {}", idx),
```

Comment:
Nit: Doesn't "Unreachable index" sound a little stange?
The rust compiler uses the following error for `array[n]`: "index out of bounds: the len is 2 but the index is 2".